src/TorchSharp/Utils/ShuffleGenerator.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+
+namespace TorchSharp.Utils
+{
+    public class ShuffleGenerator
+    {
+        int maxrange;
+        int prime;
+        int index;
+        int offset;
+        int runningvalue;
+
+        public ShuffleGenerator(int size)
+        {
+            var min = size / 2;
+            maxrange = size;
+            prime = selectCoPrimeResev(min, size);
+            offset = new Random().Next(size);
+            index = 0;
+            runningvalue = offset;
+        }
+
+        private int getCurrentValue()
```

Comment:
This seems like Java-style naming. If this is a public Pytorch-based API, then use Python names. If it is not, then please use .NET naming conventions.